src/nvim/fileio.c

```diff
@@ -4149,11 +4154,13 @@ ucs2bytes (
   return error;
 }
 
-/*
- * Return TRUE if file encoding "fenc" requires conversion from or to
- * 'encoding'.
- */
-static int need_conversion(char_u *fenc)
+/// Return true if file encoding "fenc" requires conversion from or to
+/// 'encoding'.
+///
+/// @param fenc The file encoding to check
+///
+/// @return true if conversion is required
+static bool need_conversion(char_u *fenc)
```

Comment:
FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT

Also one may change `char_u *fenc` to `const char_u *fenc` (`get_fio_flags` also needs `const char_u *` thus, but it is in the same file).